docs/rules/no-useless-escape.md

```diff
@@ -0,0 +1,49 @@
+# Disallow unnecessary escape usage (no-useless-escape)
+
+Escaping non-special characters in strings and regular expressions doesn't have any effects on results, as in the following examples:
+
+```js
+let foo = "hol\a"; // > foo = "hola"
+let bar = RegExp("hell\o"); // > bar = /hello/
```

Comment:
All of these calls to RegExp are confusing. They are still strings. Please remove them or change the function name to literally anything else.